cargo-dylint/tests/ci.rs

```diff
@@ -713,6 +715,103 @@ fn update() {
+                    // Exclude overscoped_allow as in the script
+                    if lib_name != "overscoped_allow" && lib_name != ".cargo" {
+                        restriction_libs.push(format!("--lib {lib_name}"));
+                    }
+                }
+            }
+        }
+        let restrictions_as_flags = restriction_libs.join(" ");
+
+        let mut experimental_dirs_str = Vec::new();
+        let example_experimental_dir = Path::new("examples/experimental");
+        if example_experimental_dir.is_dir() {
+            for entry in read_dir(example_experimental_dir).unwrap() {
+                let entry = entry.unwrap();
+                let path = entry.path();
+                if path.is_dir() && path.file_name().unwrap() != ".cargo" {
+                    // Convert path to string for the DIRS array
+                    experimental_dirs_str.push(path.to_string_lossy().to_string());
+                }
+            }
+        }
+
+        let base_flags =
+            format!("--lib general --lib supplementary {restrictions_as_flags} --lib clippy");
+
+        let mut dirs_to_lint = vec![
+            ".",
+            "driver",
+            "utils/linting",
+            "examples/general",
+            "examples/supplementary",
+            "examples/restriction",
+            "examples/testing/clippy",
+        ]
+        .into_iter()
+        .map(String::from)
+        .collect::<Vec<_>>();
+
+        dirs_to_lint.extend(experimental_dirs_str);
+
+        // Function similar to force_check in lint.sh
+        let workspace_root = Path::new("."); // current dir is workspace root
+        let target_dir = workspace_root.join("target");
+        if target_dir.exists() {
+            for entry in walkdir::WalkDir::new(&target_dir) {
+                let entry = entry.unwrap();
+                if entry.file_name().to_string_lossy() == ".fingerprint"
+                    && (entry
+                        .path()
+                        .starts_with(target_dir.join("dylint/target/nightly-"))
+                        || entry.path().starts_with(target_dir.join("nightly-")))
+                {
+                    let _ = remove_dir_all(entry.path());
+                }
+            }
+        }
+
+        set_var(DYLINT_RUSTFLAGS_VAR_NAME, "-D warnings");
+        eprintln!("{DYLINT_RUSTFLAGS_VAR_NAME}='-D warnings'");
+
+        for dir_path_str in &dirs_to_lint {
+            let dir_path = Path::new(dir_path_str);
+            eprintln!("Linting in directory: {dir_path:?}");
+
+            let mut cmd = Command::new(&cargo_dylint_path);
+            cmd.arg("dylint");
+            cmd.args(base_flags.split_whitespace());
+            cmd.args(["--", "--all-features", "--tests", "--workspace"]);
+            cmd.current_dir(dir_path);
+
+            // Capture output for debugging if needed
+            let output = cmd.output().expect("Failed to execute command");
```

Comment:
```suggestion
            let status = cmd.status().expect("Failed to execute command");
```
Let's just get the command's status. That way its output will be written directly to the terminal, allowing it to be seen as the command runs.